repose-aggregator/core/repose-core/src/main/java/org/openrepose/powerfilter/PowerFilter.java

```diff
@@ -398,6 +400,11 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
                         wrappedRequest.addHeader(CommonHttpHeader.TRACE_GUID.toString(),
                                 TracingHeaderHelper.createTracingHeader(traceGUID, wrappedRequest.getHeader(CommonHttpHeader.VIA.toString())));
                     }
+                    if((currentSystemModel.get().getTracingHeader() != null) &&
```

Comment:
The two `Optional` checks could probably be cleaned up a tiny bit. For example, we could do something like:
`Optional<TracingHeader> tracingHeader = Optional.ofNullable(currentSystemModel.get().getTracingHeader())`
`if (tracingHeader.map(th -> th.isEnabled()).orElse(true)) {...}`
`if (tracingHeader.map(th -> th.isSecondaryPlainText()).orElse(false)) {...}`